src/test/java/com/google/research/bleth/services/BoardParserTest.java

```diff
@@ -0,0 +1,137 @@
+package com.google.research.bleth.services;
+
+import com.google.appengine.repackaged.com.google.gson.Gson;
+import com.google.common.collect.ArrayTable;
+import com.google.research.bleth.simulator.Agent;
+import com.google.research.bleth.simulator.Board;
+import com.google.research.bleth.simulator.Location;
+import junit.framework.TestCase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BoardParserTest extends TestCase {
+
+    private static final Location zeroOnZeroCoordinate = new Location(0, 0);
+    private static final Location oneOnOneCoordinate = new Location(1, 1);
+    private static final Location zeroOnOneCoordinate = new Location(0, 1);
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+
+    @Mock
+    private Agent firstAgent;
+    @Mock
+    private Agent secondAgent;
+
+    public ArrayTable<Integer, Integer, ArrayList<String>> emptyStringListTable() {
+        ArrayTable<Integer, Integer, ArrayList<String>> table;
+        table = ArrayTable.create(IntStream.range(0, 2).boxed().collect(Collectors.toList()),
+                        IntStream.range(0, 2).boxed().collect(Collectors.toList()));
+
+        for (int row = 0; row < 2; row++) {
+            for (int col = 0; col < 2; col++) {
+                table.set(row, col, new ArrayList<String>());
+            }
+        }
+
+        return table;
+    }
+
+    public void addAgentIdInLocation (ArrayTable<Integer, Integer, ArrayList<String>> table, Agent agent, Location location) {
+        table.get(location.row, location.col).add(agent.getType() + agent.getId());
+    }
+
+    @Test
+    public void parseEmptyBoard() {
+
+        // Arrange board.
+        Board board = new Board(2, 2);
+
+        // Arrange expected serialized board.
```

Comment:
This is not the best way to test: you're verifying too much (e.g. the specific whitespace and formatting of the JSON). In effect, your test becomes a change detector - https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html

Prefer instead to parse the returned JSON, and verify that the result is *semantically* empty (i.e. all of the elements are empty lists).

The same comment applies to other tests as well.